packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -551,15 +551,21 @@ export const useGridVirtualScroller = () => {
   }, [apiRef, rowsMeta.currentPageTotalHeight]);
 
   useEnhancedEffect(() => {
-    if (enabled) {
-      // TODO a scroll reset should not be necessary
+    // TODO a scroll reset should not be necessary
+    if (enabledForColumns) {
       scrollerRef.current!.scrollLeft = 0;
+    }
+  }, [enabledForColumns, gridRootRef, scrollerRef]);
+
+  useEnhancedEffect(() => {
```

Comment:
Could we combine these two effects into one?

Reply:
Fixed https://github.com/mui/mui-x/pull/14235/commits/9b14b515320e7ca75a14afaefb9479f2cd02b77c